tmt/steps/prepare/brand_new_allmighty_install/providers/repository_provider.py

```diff
@@ -0,0 +1,141 @@
+"""
+Repository file artifact provider for tmt's allmighty-install plugin.
+
+Downloads a .repo file from a URL and places it in /etc/yum.repos.d on the guest.
+No installation support is provided, as per requirements.
+"""
+
+from collections.abc import Iterator
+from functools import cached_property
+from shlex import quote
+from typing import Any
+from urllib.parse import urlparse
+
+from tmt.container import container
+from tmt.steps.prepare.brand_new_allmighty_install.providers import (
+    ArtifactInfo,
+    ArtifactProvider,
+    ArtifactType,
+    DownloadError,
+)
+from tmt.steps.provision import Guest
+from tmt.utils import GeneralError, Path, ShellScript
+
+
+@container
+class RepositoryArtifactInfo(ArtifactInfo):
```

Comment:
I thought that `ArtifactInfo` is supposed to be used for information about the artifacts, e.g. `rpms`, not information about the delivery method, e.g. repo file. Let's clarify these on the review session?